app/assets/javascripts/omni-nav-v2.js

```diff
@@ -1,38 +1,109 @@
-var omni_nav_v2 = {
+var OmniNav2 = (function() {
 
-  initialize: function() {
-    this.$container = $('#omni-nav-v2');
-    this.$utility_nav = this.$container.find('.utility-nav');
-    this.$primary_nav = this.$container.find('#primary-nav');
 
+  var $container, $utility_nav, $primary_nav;
 
-    $('.utility-nav-trigger').on('click', omni_nav_v2.onUtilityNavClick);
-    this.$utility_nav.find('li.utility-has-dropdown').on('click', omni_nav_v2.onUtilityNavDropdownClick);
+  var initialize = function(container, utility_nav, primary_nav) {
+    $container = container;
+    $utility_nav = utility_nav;
+    $primary_nav = primary_nav;
+    $('.utility-nav-trigger').on('click', onUtilityNavClick);
+    $utility_nav.find('li.utility-has-dropdown').on('click', onUtilityNavDropdownClick);
+    $utility_nav.find('input:text').on('input', onSearchInput);
+    $primary_nav.find('input:text').on('input', onSearchInput); 
+  }
 
-  },
+  var onUtilityNavClick = function() {
+    $('.primary-nav-action').toggleClass("utility-open");
 
-  onUtilityNavClick : function() {
     if($(window).width() >= 768 ){
```

Comment:
768 is a magic number here. Can we set it as a constant?

Also, I prefer this:

    if ( $(window).width() >= 768 ) {

To this:

    if($(window).width() >= 768 ){

Does the style guide we reference express a preference? Let me check ...

According to [this](https://github.com/airbnb/javascript#blocks), it seems to recommend:

    if ($(window).width() >= 768) {